### PR TITLE
KeyVault support for a Nuget Credential provider (.NET Core)

### PR DESCRIPTION
**Pull request by denislevin**
This allows to configure the provider to use Azure KeyVault to store PAT tokens for Nuget feeds.
For cloud deployments with multiple on demand VMs, all that needs to be set is a KeyVault URL in the config file.
The configuration supports authentication to KeyVault using MSI (Managed Server Identity) or via a certificate installed in the user or machine store. In the second case, authentication is to the AzureID application identified by cliendId, and that application should have access to KeyVault.

### CODE REVIEW

File: CredentialProvider.Microsoft/CredentialProviders/KeyVault/KeyVaultHelper.cs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    /// </summary>
+    public sealed class KeyVaultHelper : IDisposable
+    {
+        public struct Config
+        {
+            public string CertificateThumbprint { get; set; }
+            public string CertificateStoreType { get; set; }
+            public string KeyVaultUrl { get; set; }
+            public string ClientId { get; set; }
+            public bool? UseMsi { get; set; }
```
Comment (loic-sharma) [inline CredentialProvider.Microsoft/CredentialProviders/KeyVault/KeyVaultHelper.cs]: Is this a customer facing config? I would consider expanding this acronym considering it's not all that common.

Comment (denislevin) [reply CredentialProvider.Microsoft/CredentialProviders/KeyVault/KeyVaultHelper.cs]: This is not an external data structure unless one wants to reuse KeyVaultHelper class with a different project. I can make it internal. Customer-facing config is in ConfigManager, string keys for key-value pairs in the config. 
I use the following values:
KeyVaultUrl
KeyVaultUseMsi
KeyVaultAuthClientId
KeyVaultAuthCertificateStoreType
KeyVaultAuthCertificateThumbprint

I can make KeyVaultUseMsi implicit when CertificateThumbprint is not specified, or can expand it to KeyVaultUseManagedServerIdentity.

**Comment by denislevin** [comment]
Is there any interest to getting this PR? We've been using this functionality for 2 years, seems stable

**Review by Thiri25 — COMMENTED** [review]
.vsts-hosted.yml

**Comment by Thiri25** [comment]
.vsts-hosted.yml

File: CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs [hunk 2]
```diff
@@ -123,7 +123,12 @@ public override async Task<GetAuthenticationCredentialsResponse> HandleRequestAs
...
-            return null;
+            return new GetAuthenticationCredentialsResponse(
+                null,
+                null,
+                message: null,
+                authenticationTypes: new List<string>() { "Basic" },
+                responseCode: MessageResponseCode.NotFound);
         }
     }
 }
```
Comment (LuckyLuv77-PC) [inline CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsCredentialProvider.cs]: Laura Vanlaningham, (atlslscsrv-app@atlsecsrv.onmicrosoft.com)

**Comment by phil-hodgson** [comment]
Closing. There doesn't appear to be broad support for taking this change at this time. Thank you for your contribution.

**Pull request closed without merging**